lua/lib/lib.lua

```diff
@@ -366,4 +367,16 @@ function M.toggle_dotfiles()
   return M.refresh_tree()
 end
 
+function M.dir_up(node)
+  if not node then
+    return M.change_dir('..')
+  else
+    local newdir = vim.fn.fnamemodify(node.absolute_path, ':h')
+    if newdir == M.Tree.cwd then
+      M.change_dir('..')
+    end
+    return M.set_index_and_redraw(newdir)
```

Comment:
I find it confusing when going up a dir it closes the expanded path to the current node and jumps the cursor to the top of the tree, i'd rather see it this way
```suggestion
    local newdir = vim.fn.fnamemodify(M.Tree.cwd, ':h')
    if newdir == M.Tree.cwd then
      M.change_dir('..')
    end
    return M.set_index_and_redraw(node.absolute_path)
```
what do you think?

Reply:
Yes, I like that better. I got rid of the `if newdir..end` part as that was blocking anything from happening.

One small problem with this approach is that there is no way to navigate back down to the current working directory, so I also added a small patch to `lib.open` to change directories back to the invoking window's cwd when the tree window is reopened. I tried a few approaches to accomplish this, and this seemed to work the best, but certainly open to input!

Thanks for considering!